include/ppx/knob.h

```diff
@@ -314,6 +316,67 @@ class KnobDropdown final
     std::vector<T> mChoices;
 };
 
+// KnobConstant is intended for parameters that cannot be adjusted when the application is run
+// They will not be displayed in the UI and cannot have their values changed by the application
+template <typename T>
+class KnobConstant final
+    : public Knob
+{
+public:
+    KnobConstant(const std::string& flagName, T defaultValue)
+        : Knob(flagName, false)
+    {
+        SetValue(defaultValue);
+    }
+
+    KnobConstant(const std::string& flagName, T defaultValue, T minValue, T maxValue)
+        : Knob(flagName, false)
+    {
+        static_assert(std::is_arithmetic_v<T>, "KnobConstant can only be defined with min/max when it's of arithmetic type");
+        PPX_ASSERT_MSG(minValue < maxValue, "invalid range to initialize KnobConstant");
+        PPX_ASSERT_MSG(minValue <= defaultValue && defaultValue <= maxValue, "defaultValue is out of range");
+
+        SetValidator([minValue, maxValue](T newValue) {
+            if (newValue < minValue || newValue > maxValue) {
+                return false;
+            }
+            return true;
+        });
+
+        SetValue(defaultValue);
+    }
+
+    void Draw() override {} // KnobConstant is invisible in the UI
```

Comment:
Should we add an assert in the functions that shouldn't be called? like `Draw`, `ResetToDefault` or is it valid to call them and expect nothing happens?